src/ucp/core/ucp_ep.c

```diff
@@ -1036,27 +1036,30 @@ void ucp_ep_destroy(ucp_ep_h ep)
 }
 
 static int
-ucp_ep_config_lane_is_dst_rsc_index_equal(const ucp_ep_config_key_t *key1,
-                                          ucp_lane_index_t lane1,
-                                          const ucp_ep_config_key_t *key2,
-                                          ucp_lane_index_t lane2)
+ucp_ep_config_lane_is_dst_index_equal(ucp_rsc_index_t dst_index1,
+                                      ucp_rsc_index_t dst_index2)
 {
-    return /* at least one of destination resource index is not specified */
-           (key1->lanes[lane1].dst_rsc_index == UCP_NULL_RESOURCE) ||
-           (key2->lanes[lane2].dst_rsc_index == UCP_NULL_RESOURCE) ||
-           /* both destination resource index are the same */
-           (key1->lanes[lane1].dst_rsc_index == key2->lanes[lane2].dst_rsc_index);
+    return /* at least one of destination indicies is not specified */
+           (dst_index1 == UCP_NULL_RESOURCE) ||
+           (dst_index2 == UCP_NULL_RESOURCE) ||
+           /* both destination indicies are the same */
```

Comment:
minor: i'd avoid such comments, as it just repeats what code does..

Reply:
I just fixed the previous comments to be suitable in the new code
this comment should be read together with the above to make the code clear
but I'm ok ti remove if any other comments